Dockerfile.alpine

```diff
@@ -38,7 +38,7 @@ COPY . .
 RUN CGO_ENABLED=0 GOOS=linux go install . ./cmd/...
 
 # Production image
-FROM alpine:3.14.0
+FROM alpine@sha256:adab3844f497ab9171f070d4cae4114b5aec565ac772e2f2579405b78be67c96
```

Comment:
Similarly, this does not match the official image for alpine 3.14 (which is at 3.14.2) for me.

Reply:
```
➜  dcrd git:(naveen/feat/fix-docker-sha) crane digest alpine:3.14.0
sha256:adab3844f497ab9171f070d4cae4114b5aec565ac772e2f2579405b78be67c96
```

Reply:
> Similarly, this does not match the official image for alpine 3.14 (which is at 3.14.2) for me.

I kept the original image to  `alpine:3.14.0`

Reply:
```
$ docker pull alpine:3.14.0
...
Digest: sha256:adab3844f497ab9171f070d4cae4114b5aec565ac772e2f2579405b78be67c96
```

Confirming that `adab3844f497ab9171f070d4cae4114b5aec565ac772e2f2579405b78be67c96` matches what docker pull retrieves.